lib/item.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+
+/// Item:
+///   - objects of this class store a single item
+///   in a list (needs to be wrapped in an ItemWidget
+///   to be displayed)
+class Item {
+  // I would like to know your thoughts on this next line.
+  final TextEditingController textEditingController;
+  // I directly stored a TextEditingController in each Item
+  // instead of in each ItemWidget mainly for this reason:
+  //   - The TextEditingController already stores the value
+  //     of each item. So, if this controller were moved to
+  //     the ItemWidget, the value of the item would be stored
+  //     directly in the ItemWidget, meaning we would either
+  //     have to store ItemWidgets or TextEditingControllers
+  //     in listItems (since those would be what contained the
+  //     text value of each item). It would be possible to store
+  //     Strings or Items in listItems (instead of ItemWidgets).
+  //     Then, each TextEditingController would be stored in an
+  //     ItemWidget, constantly updating the value of its
+  //     corresponding item. I think this would be redundant and
+  //     add additional complexity.
+  // However, my solution seems to have a "code smell." What do you
```

Comment:
You are correct, this isn't a good design pattern. This class is meant for storing data, but it contains a widget.  The instances of this class are part of the "model" of your app. You don't want to store widgets in an object that is part of the model. Widgets are part of the "view" of your app. You want to keep the model and view separated where you can.  Skim over [this article](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) for some background.

Reply:
The most important thing to do is to remove the `TextEditingController` from the `Item` class. I left two possible solutions below.